libero/src/main/groovy/com/connexta/libero/Libero.groovy

```diff
@@ -56,49 +82,93 @@ class Libero {
                     println "Response must be either [y/n]"
                     System.exit(1)
             }
-        }
-        else {
+        } else {
             executeRelease(config)
         }
+        git.checkout(branch: originalBranch)
+    }
+
+    /**
+     * Checks config file for missing values prior to checkout
+     * @param config
+     */
+    private void initializeConfig(Config config) {
+        config.projectName = config.projectName ?: new File(config.projectDir).getName()
+        config.sourceRemote = config.sourceRemote ?: 'origin'
+        config.destBranch = config.destRemote ?: config.sourceRemote
+        // might need to default to something like "refs/remotes/${config.sourceRemote}/master"
+        config.ref = config.ref ?: 'master'
+        config.destBranch = config.destBranch ?: 'master'
+        config.preProps = config.preProps ?: null as Map
+        config.postProps = config.postProps ?: null as Map
+        config.gitPush = config.gitPush ?: false
+        config.quickBuild = config.quickBuild ?: false
+        config.dryRun = config.dryRun ?: false
+        config.force = config.force ?: false
+    }
+
+    /**
+     * performs initial preparation of git repository
+     * @param config
+     * @param git
+     */
+    private void initialRepoPreparation(Config config, Grgit git) {
+        git.fetch(remote: config.sourceRemote)
+        git.checkout(branch: config.ref)
+        git.pull(remote: config.sourceRemote)
```

Comment:
Doesn't `pull` do a `fetch` as well? Couldn't we do `git.pull(...)` and then `git.checkout(...)`?